src/rust/bitbox02-rust/src/hww/api/system.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::Error;
+use crate::pb;
+
+use pb::reboot_request::Purpose;
+use pb::response::Response;
+
+use crate::workflow::confirm;
+
+pub async fn reboot(&pb::RebootRequest { purpose }: &pb::RebootRequest) -> Result<Response, Error> {
+    confirm::confirm(&confirm::Params {
+        title: "",
+        body: match Purpose::from_i32(purpose) {
+            Some(Purpose::Upgrade) => "Proceed to upgrade?",
+            Some(Purpose::Settings) => "Go to\nstartup settings?",
+            // No error, if new client library sends a purpose that we don't understand,
+            // we reboot anyway.
+            None => "Reboot?",
```

Comment:
This is what will now be displayed if the host doesn't provide a purpose, right. What I imagine is

- a user sees "firmware upgrade available" in the app
- clicks on "upgrade"
- presented with "reboot?" on the device display
- possibly confused what reboot has to do with upgrade?

How about `None` means "Proceed to upgrade?" to keep the original behaviour and add only `Purpose::Settings`?

Reply:
Ah wait, an earlier commit says "Upgrade device (default, backwards compatible)". Where's the default set?

BTW, I'd squash all or most commits.

Reply:
Oh, never mind. Somehow I missed that `UPGRADE` is `0`.

Reply:
Exactly, 0 is default. I simply made unknown values also reboot instead of returning an error, just in case.